src/hb-font.h

```diff
@@ -530,6 +513,42 @@ typedef void (*hb_font_get_glyph_shape_func_t) (hb_font_t *font, void *font_data
 						hb_draw_funcs_t *draw_funcs, void *draw_data,
 						void *user_data);
 
+/**
+ * hb_font_draw_glyph_func_t:
+ * @font: #hb_font_t to work upon
+ * @font_data: @font user data pointer
+ * @glyph: The glyph ID to query
+ * @draw_funcs: The draw functions to send the shape data to
+ * @draw_data: The data accompanying the draw functions
+ * @user_data: User data pointer passed by the caller
+ *
+ * Alternative name for #hb_font_get_glyph_shape_func_t.
+ *
+ * Since: REPLACEME
+ *
+ **/
+typedef void (*hb_font_draw_glyph_func_t) (hb_font_t *font, void *font_data,
+                                           hb_codepoint_t glyph,
+                                           hb_draw_funcs_t *draw_funcs, void *draw_data,
+                                           void *user_data);
+
```

Comment:
IMHO again, I don’t believe the new names (even if we decide they are better, which I’m not convinced about) are so better it is worth the duplication and the confusing and the complications of API deprecation.

Reply:
I mean the client will have to check for HarfBuzz versions and call different functions for different versions and all these complications, but it is the exact same API only renamed.

Reply:
Yes, its a tradeoff. Between short-term inconvenience and long-term inconsistency

Reply:
Nobody is forced to do anything anyway. The old api is not going away

Reply:
The api is called hb-draw anyway, so...

Reply:
The only argument I see here is whether the new names are better or not.

And yes, at the very beginning, it was called hb_font_get_glyph_paint. But I find that a pretty awkward name, even worse than hb_font_get_glyph_shape, which should probably have been called hb_font_glyph_outline, with hb_outline_funcs_t. Too late now.

Reply:
I understand the hesitation but I like the consistency of the new API.

Reply:
The experimental API had hb_font_draw_glyph, and it wad renamed when it got stabalized last February, now we are renaming it back...

Reply:
I feel like there's few enough users that we can update all.

Reply:
I'll try remain neutral here and leave this decision up to you. My input is that it should be consistent.

Either get_glyph_shape+get_glyph_paint or draw_glyph+paint_glyph. A mixture is worth either of these alternatives.

Reply:
@khaledhosny Would not deprecating the old `hb_font_get_glyph_shape` name change anything for you?

Reply:
It would at least keep existing clients undisturbed. Having two sets of identical but differently named functions will be confusing, but if you feel strongly about the new names, we cn probably hide the old ones from the documentation.

Reply:
I want an API that has draw and paint in parallel in it.  I don't feel as strongly as you do about deprecation. Specially since we control all users of this API currently. I doubt anyone else is using it other than FontGoggles and uharfbuzz.

Reply:
Plus, we keep deprecated API forever...

Reply:
> I doubt anyone else is using it other than FontGoggles and uharfbuzz.

I’m using it in (to be released) LibreOffice 7.5. If the old API gets deprecated, I’ll have to do some cpp dance around it since packagers don’t like deprecation warnings, not a show stopper but feel unnecessary. Anyway, I don’t want to block this on a naming issue, so your call.

Reply:
I like the new naming so if you don't mind I like to go ahead with it.

Reply:
The new naming then.